src/controllers/productController.ts

```diff
@@ -118,4 +119,28 @@ export default class productController {
       return res.status(500).json({ message: 'Internal server error' })
     }
   }
+
+  /**
+   * delete product
+   * @param {Request} req - Express request object
+   * @param {Response} res - Express response object
+   */
+  static async deleteProduct(req: Request, res: Response) {
+    try {
+      const productId = req.params.productId;
+      const sellerId = req.user?.id
+      const product = await getProductById(sellerId, productId)
+      if(product){
+        const result: number = await deleteProductById(sellerId, productId)
+        if (result !== 0) {
+          return res.status(200).json({message: "item deleted"})
+        }
+      } else {
+        return res.status(410).json({message: "item not found"});
```

Comment:
```suggestion
        return res.status(404).json({message: "item not found"});
```
 check  if this is correct  status code